### PR TITLE
Delayed creating resource instances

### PR DESCRIPTION
**Pull request by anniezhang77**
risk: low

### CODE REVIEW

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala [hunk 1]
```diff
@@ -53,68 +55,74 @@ object ResourceMgr {
...
-            instId
-          )
-        } yield running(ps, rscInst, instId)
+      // here we make all invalid input to default 8 hours, the input should do validation before
+      // saving them to DB
+      val terminateAfterDuration: FiniteDuration =
+      try {
+        (resourceR.terminateAfter * 1.hour).asInstanceOf[FiniteDuration]
+      }
+      catch {
```
Comment (realstraw) [inline orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala:70]: can you join this with the previous line?

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala [hunk 2]
```diff
@@ -126,6 +134,11 @@ object ResourceMgr {
         ps.ctx.log.info(s"${ps.ctx.self} (idle) received GetResourceInstSpec($replyTo)")
         ps.database.sync(ps.resourceQuery.setRun())
         val instId = 1
+        ps.ctx.self ! CreateResourceInst(replyTo, instId)
+        Behaviors.same
```
Comment (realstraw) [inline orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala:138]: what if `idle` receives `CreateResourceInst` again?

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala [hunk 3]
```diff
@@ -167,6 +180,17 @@ object ResourceMgr {
         ps.ctx.log.info(s"${ps.ctx.self} (running) received GetResourceInstSpec($replyTo)")
         resourceInst ! ResourceInstance.GetResourceInstSpec(replyTo)
         Behaviors.same
+      case CreateResourceInst(replyTo, instId) =>
```
Comment (realstraw) [inline orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala:183]: why would resource in `running` state also be able to create new instances?

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala (hunk 2 above)
Comment (anniezhang77) [reply orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala:138]: after it received the first `CreateResourceInst` it'll become running or finished state. It won't go back to idle anymore.

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala (hunk 1 above)
Comment (anniezhang77) [reply orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala]: done

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala (hunk 3 above)
Comment (anniezhang77) [reply orchard-core/src/main/scala/com/salesforce/mce/orchard/system/actor/ResourceMgr.scala:182]: good point, updated

**Pull request merged**